solvebio/resource/object.py

```diff
@@ -484,24 +486,34 @@ def is_file(self):
         return self.object_type == 'file'
 
     def has_tag(self, tag):
-        """Return True if object contains tags"""
+        """Return True if object contains tag"""
 
         def lowercase(x):
             return x.lower()
 
-        return lowercase(tag) in map(lowercase, self.tags)
+        return lowercase(str(tag)) in map(lowercase, self.tags)
 
-    def tag(self, tags, remove=False, dry_run=False, apply_save=False):
+    def tag(self, tags, remove=False, dry_run=False, apply_save=True):
         """Add or remove tags on an object"""
 
+        def iterable(arg):
```

Comment:
rename to `is_iterable_non_string` and add a docstring saying its a py2or3 compat way to check